tests/test_nativetypes.py

```diff
@@ -122,6 +128,18 @@ def test_string_top_level(env):
     assert result == "Jinja"
 
 
+def test_string_concatenation(async_native_env):
+    async def async_render():
+        t = async_native_env.from_string(
+            "{%- macro x(y) -%}{{ y }}{%- endmacro -%}{{- x('not') }} {{ x('bad') -}}"
+        )
+        result = await t.render_async()
+        assert isinstance(result, str)
+        assert result == "not bad"
+
+    asyncio.run(async_render())
```

Comment:
this should use

```python
def _asyncio_run(async_fn, *args):
    return asyncio.run(async_fn(*args))


@pytest.fixture(params=[_asyncio_run, trio.run], ids=["asyncio", "trio"])
def run_async_fn(request):
    return request.param


def test_string_concatenation(async_native_env, run_async_fn):
    async def async_render():
        t = async_native_env.from_string(
            "{%- macro x(y) -%}{{ y }}{%- endmacro -%}{{- x('not') }} {{ x('bad') -}}"
        )
        result = await t.render_async()
        assert isinstance(result, str)
        assert result == "not bad"

    run_async_fun(async_render)
```

Reply:
Thanks!  That suggested `run_async_fn` fixture is used in two locations at the moment (in the `main` branch of the Jinja test suite, that is) - perhaps it'd make sense to promote it to be a shared `conftest.py` fixture?

Reply:
> Thanks! That suggested `run_async_fn` fixture is used in two locations at the moment (in the `main` branch of the Jinja test suite, that is) - perhaps it'd make sense to promote it to be a shared `conftest.py` fixture?

I've pushed a few changes that make that suggestion: re-using a single definition of a `run_async_fn` fixture from `conftest.py`.